Sources/CCryptoBoringSSL/CMakeLists.txt

```diff
@@ -0,0 +1,322 @@
+  crypto/pkcs8/p5_pbev2.c
+  crypto/pkcs8/pkcs8.c
+  crypto/pkcs8/pkcs8_x509.c
+  crypto/poly1305/poly1305.c
+  crypto/poly1305/poly1305_arm.c
+  crypto/poly1305/poly1305_vec.c
+  crypto/pool/pool.c
+  crypto/rand_extra/deterministic.c
+  crypto/rand_extra/forkunsafe.c
+  crypto/rand_extra/fuchsia.c
+  crypto/rand_extra/rand_extra.c
+  crypto/rand_extra/windows.c
+  crypto/rc4/rc4.c
+  crypto/refcount_c11.c
+  crypto/refcount_lock.c
+  crypto/rsa_extra/rsa_asn1.c
+  crypto/rsa_extra/rsa_print.c
+  crypto/siphash/siphash.c
+  crypto/stack/stack.c
+  crypto/thread.c
+  crypto/thread_none.c
+  crypto/thread_pthread.c
+  crypto/thread_win.c
+  crypto/trust_token/pmbtoken.c
+  crypto/trust_token/trust_token.c
+  crypto/trust_token/voprf.c
+  crypto/x509/a_digest.c
+  crypto/x509/a_sign.c
+  crypto/x509/a_strex.c
+  crypto/x509/a_verify.c
+  crypto/x509/algorithm.c
+  crypto/x509/asn1_gen.c
+  crypto/x509/by_dir.c
+  crypto/x509/by_file.c
+  crypto/x509/i2d_pr.c
+  crypto/x509/rsa_pss.c
+  crypto/x509/t_crl.c
+  crypto/x509/t_req.c
+  crypto/x509/t_x509.c
+  crypto/x509/t_x509a.c
+  crypto/x509/x509.c
+  crypto/x509/x509_att.c
+  crypto/x509/x509_cmp.c
+  crypto/x509/x509_d2.c
+  crypto/x509/x509_def.c
+  crypto/x509/x509_ext.c
+  crypto/x509/x509_lu.c
+  crypto/x509/x509_obj.c
+  crypto/x509/x509_r2x.c
+  crypto/x509/x509_req.c
+  crypto/x509/x509_set.c
+  crypto/x509/x509_trs.c
+  crypto/x509/x509_txt.c
+  crypto/x509/x509_v3.c
+  crypto/x509/x509_vfy.c
+  crypto/x509/x509_vpm.c
+  crypto/x509/x509cset.c
+  crypto/x509/x509name.c
+  crypto/x509/x509rset.c
+  crypto/x509/x509spki.c
+  crypto/x509/x_algor.c
+  crypto/x509/x_all.c
+  crypto/x509/x_attrib.c
+  crypto/x509/x_crl.c
+  crypto/x509/x_exten.c
+  crypto/x509/x_info.c
+  crypto/x509/x_name.c
+  crypto/x509/x_pkey.c
+  crypto/x509/x_pubkey.c
+  crypto/x509/x_req.c
+  crypto/x509/x_sig.c
+  crypto/x509/x_spki.c
+  crypto/x509/x_val.c
+  crypto/x509/x_x509.c
+  crypto/x509/x_x509a.c
+  crypto/x509v3/pcy_cache.c
+  crypto/x509v3/pcy_data.c
+  crypto/x509v3/pcy_lib.c
+  crypto/x509v3/pcy_map.c
+  crypto/x509v3/pcy_node.c
+  crypto/x509v3/pcy_tree.c
+  crypto/x509v3/v3_akey.c
+  crypto/x509v3/v3_akeya.c
+  crypto/x509v3/v3_alt.c
+  crypto/x509v3/v3_bcons.c
+  crypto/x509v3/v3_bitst.c
+  crypto/x509v3/v3_conf.c
+  crypto/x509v3/v3_cpols.c
+  crypto/x509v3/v3_crld.c
+  crypto/x509v3/v3_enum.c
+  crypto/x509v3/v3_extku.c
+  crypto/x509v3/v3_genn.c
+  crypto/x509v3/v3_ia5.c
+  crypto/x509v3/v3_info.c
+  crypto/x509v3/v3_int.c
+  crypto/x509v3/v3_lib.c
+  crypto/x509v3/v3_ncons.c
+  crypto/x509v3/v3_ocsp.c
+  crypto/x509v3/v3_pci.c
+  crypto/x509v3/v3_pcia.c
+  crypto/x509v3/v3_pcons.c
+  crypto/x509v3/v3_pmaps.c
+  crypto/x509v3/v3_prn.c
+  crypto/x509v3/v3_purp.c
+  crypto/x509v3/v3_skey.c
+  crypto/x509v3/v3_utl.c)
+
+target_include_directories(CCryptoBoringSSL PUBLIC
+  include)
+  
+set_target_properties(CCryptoBoringSSL PROPERTIES
+  INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
```

Comment:
I think that we need to set `WIN32_LEAN_AND_MEAN` as the generated files for boring ssl cause issues otherwise.  Something like this should do the trick:

```cmake
target_compile_definitions(CCryptoBoringSSL PRIVATE
  $<$<PLATFORM_ID:Windows>:WIN32_LEAN_AND_MEAN>)
```